test/mainnet-fork/Approve.t.sol

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity >=0.8.0;
+
+import "./ForkTestBase.t.sol";
+
+import { ForeignController } from "../../src/ForeignController.sol";
+import { MainnetController } from "../../src/MainnetController.sol";
+
+contract MainnetControllerHarness is MainnetController {
+
+    constructor(
+        address admin_,
+        address proxy_,
+        address rateLimits_,
+        address vault_,
+        address psm_,
+        address daiUsds_,
+        address cctp_
+    ) MainnetController(admin_, proxy_, rateLimits_, vault_, psm_, daiUsds_, cctp_) {}
+
+    function approve(address token, address spender, uint256 amount) external {
+        _approve(token, spender, amount);
```

Comment:
Need to have an additional harness function where we call the CurveLib.approve

Reply:
And then should have a `_approveCurveTest` and another mainnet controller test for it

Reply:
added the tests